common/src/main/java/com/wynntils/features/user/overlays/InfoBoxFeature.java

```diff
@@ -0,0 +1,159 @@
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.resources.language.I18n;
+
+@FeatureInfo(category = FeatureCategory.OVERLAYS)
+public class InfoBoxFeature extends UserFeature {
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox1Overlay = new InfoBoxOverlay(1);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox2Overlay = new InfoBoxOverlay(2);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox3Overlay = new InfoBoxOverlay(3);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox4Overlay = new InfoBoxOverlay(4);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox5Overlay = new InfoBoxOverlay(5);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox6Overlay = new InfoBoxOverlay(6);
+
+    public static class InfoBoxOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public String content = "";
+
+        @Config
+        public float secondsPerRecalculation = 0.5f;
+
+        private final int id;
+        private final List<Function<?>> functionDependencies = new ArrayList<>();
+        private TextRenderTask renderTask;
+        private long lastUpdate = 0;
+
+        private TextRenderTask getRenderTask(String content) {
+            return FunctionManager.getStringFromLegacyTemplate(content)
+                    .setSetting(TextRenderSetting.DEFAULT
+                            .withHorizontalAlignment(this.getRenderHorizontalAlignment())
+                            .withMaxWidth(this.getWidth())
+                            .withCustomColor(CustomColor.fromChatFormatting(ChatFormatting.WHITE))
+                            .withTextShadow(textShadow));
+        }
+
+        public InfoBoxOverlay(int id) {
+            super(
+                    new OverlayPosition(
+                            -60 + (15 * id),
+                            5,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.MiddleLeft),
+                    new GuiScaledOverlaySize(120, 10),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle);
+            this.id = id;
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WynnUtils.onWorld()) return;
+
+            if (System.nanoTime() - lastUpdate > secondsPerRecalculation * 1e+9) {
```

Comment:
@magicus I hope you do not mind this. Overlays are not registered to the event bus... This is a good enough replacement IMO.

Reply:
Maybe do the same to preview?

Reply:
Yeah, but ticking 6 different overlays is pretty messy from a feature event.

Reply:
I think it is fine for previews to be updated every time, for now. If there comes some really heavy variable, we can update it. (also, the active function by nature uses some caching to prevent per render call calculation)

Reply:
Ok, fine. (We should perhaps also let the preview use hard-coded functions from us, since if the user has weird strings it might now show up properly. But that's a future improvement, if we should do it at all.)